db/sovereign-db/src/state_db.rs

```diff
@@ -17,6 +17,7 @@ use crate::{
 #[derive(Clone)]
 pub struct StateDB {
     db: Arc<DB>,
+    version: Version,
```

Comment:
It think this needs to be an `Arc<Mutex<Version>>` since we clone the StateDB everywhere

Reply:
This is a good catch! We only increment the version in the `Storage::finalize,` and only the top-level version of the storage calls it, that's why the tests didn't catch it. Updated it to use Mutex so we are safe in the future.